### ISSUE TITLE
LayerWiseConfigurations are not equal after restoring a CNN using ModelSerializer

### ISSUE THREAD

**Issue by magorismagor** [deeplearning4j/deeplearning4j#3015]
Version 0.7.0.

I am training a CNN, and saving and restoring the model using the ModelSerializer. However, although the json from the saved and the restored networks are equal, their MultiLayerConfiguration are not equal. I attach the saved zip.
[modelAfter.zip](https://github.com/deeplearning4j/deeplearning4j/files/833956/modelAfter.zip)

This is the simple code:

ModelSerializer.writeModel(model, "modelAfter.zip", true);
MultiLayerNetwork restoredModel = ModelSerializer.restoreMultiLayerNetwork("modelAfter.zip", true);
 System.out.println("Saved and loaded parameters are equal:      " + model.params().equals(restoredModel.params()));
            System.out.println("Saved and loaded configurations are equal:  " + model.getLayerWiseConfigurations().equals(restoredModel.getLayerWiseConfigurations()));

This is the output:

Saved and loaded parameters are equal:      true
Saved and loaded configurations are equal:  false

**Comment by agibsonccc** [comment]
Mind using the latest version first?

**Comment by magorismagor** [comment]
I will check that with the new 0.8.0 version, however this is a minor bug as the parameters are loaded correctly as @raver119 already said.

**Comment by AlexDBlack** [comment]
I'm unable to reproduce this on current master.
@magorismagor I took your configuration.json file then ran the code that you provided. I also tested the LeNet example config. Here's what I'm getting:

```
Saved and loaded parameters are equal: true
Saved and loaded configurations are equal: true
```

```java
        File f = new File("configuration.json");
        String json = FileUtils.readFileToString(f);
        MultiLayerConfiguration conf = MultiLayerConfiguration.fromJson(json);

        MultiLayerNetwork model = new MultiLayerNetwork(conf);
        model.init();

        ModelSerializer.writeModel(model, "modelAfter.zip", true);
        MultiLayerNetwork restoredModel = ModelSerializer.restoreMultiLayerNetwork("modelAfter.zip", true);
        System.out.println("Saved and loaded parameters are equal: " + model.params().equals(restoredModel.params()));
        System.out.println("Saved and loaded configurations are equal: " + model.getLayerWiseConfigurations().equals(restoredModel.getLayerWiseConfigurations()));
```

I'll close this. We can re-open if you can reproduce this with 0.8.0

**Comment by magorismagor** [comment]
The same issue in 0.8.0, however I checked that the comparison returns false only after performing some training in one of the networks, and true if you does not perform any training. Just to clarify:

1.Create the network (model1)
2.Train for one or more epochs
3.Save the model1
4.Load the saved model1 in model2
5.layerWiseConfiguration comparison of model1 and model2 returns false
6.train model2 for one or more epochs
7.save the model2
8.load the saved model2 in model3
9.layerWiseConfiguration comparison of model2 and model3 returns false
10.save model3
11.load model3 in model4
12.layerWiseConfiguration comparison of model3 and model 4 returns true

Anyway, as the configuration is loaded correctly, I suggest to implement the method "equals" in the class MultiLayerConfiguration to compare for example the result of the output of the toJson() method. I checked this comparison and returns true in all the previous cases.

Something like:

public boolean equals(MultiLayerConfiguration confToCompare) {
         return StringUtils.equals(toJson(),confToCompare.toJson());  
 }

Other option is to modify the example where this comparison is shown, to use the comparison of the output of the toJson() methods of the two configurations instead using the equals method, as in the present example.
